src/bin/bench.rs

```diff
@@ -185,19 +187,40 @@ fn perform_tree_operations(tree: Tree, config: Config) {
         (Op::Delete, config.delete),
         (Op::Cas, config.cas),
     ];
-
-    // TODO update to handle division errors
-    let ops_per_thread = config.num_operations / config.num_threads;
+    let ops_per_thread = config.num_operations / config.num_threads; // TODO update to handle division errors
     let mut threads = Vec::new();
+    let ops_per_second = AtomicUsize::new(0);
+    let ops_per_second_vec = Vec::new();
+    let mut statistics = Statistics::new();
+
     let tree = Arc::new(tree);
     let ops = Arc::new(ops);
+    let ops_per_second = Arc::new(ops_per_second);
+    let ops_per_second_vec = Arc::new(Mutex::new(ops_per_second_vec));
+
+    // thread which spits out bench-related results every 1 second
+    {
+        let ops_per_second = ops_per_second.clone();
+        let ops_per_second_vec = ops_per_second_vec.clone();
+        thread::spawn(move || {
+            loop {
+                thread::sleep(Duration::from_secs(1));
+                let ops_per_second = ops_per_second.swap(0, Ordering::Relaxed);
+                let mut ops_per_second_vec = ops_per_second_vec.lock().unwrap();
```

Comment:
I think we can probably just print out the throughput each second as it happens, rather than buffering it

Reply:
Good idea. This makes the code easier to write for me 😄!

So we get rid of the array to store all the throughputs and won't have an overall throughput after the benchmark is done?